packages/shared/src/hooks/useProfileTooltip.ts

```diff
@@ -0,0 +1,39 @@
+import request from 'graphql-request';
+import { useMemo, useState } from 'react';
+import { useQuery } from 'react-query';
+import {
+  UserTooltipContentData,
+  USER_TOOLTIP_CONTENT_QUERY,
+} from '../graphql/users';
+import { apiUrl } from '../lib/config';
+
+interface UseProfileTooltip {
+  fetchInfo: () => unknown;
+  data?: UserTooltipContentData;
+}
+
+export const useProfileTooltip = (userId: string): UseProfileTooltip => {
```

Comment:
I'd like to suggest a different design instead of `fetchInfo`. You should provide another argument `enabled: boolean` to the hook and then you can provide it to the `enabled` prop of react query.
The current design seems somewhat like an anti pattern. What do you think?

Reply:
Was actually thinking about that, what made me do it this way is, that we won't have to manage an additional state outside for every consumer utilizing this hook.

Reply:
Ok then. Let's keep it this way :)